v0/destinations/ProfitWell/transform.js

```diff
@@ -0,0 +1,195 @@
+    get(message, "context.traits.subscriptionAlias");
+
+  if (!subscriptionId && !subscriptionAlias) {
+    throw new CustomError(
+      "subscriptionId or subscriptionAlias is required for identify",
+      400
+    );
+  }
+
+  const targetUrl = `${baseEndpoint}/v2/users/${userId || userAlias}`;
+  const res = await getSubscriptionHistory(targetUrl, {
+    headers: {
+      Authorization: Config.privateApiKey
+    }
+  });
+
+  let payload;
+  const response = defaultRequestConfig();
+
+  if (res.success) {
+    // some() breaks if the callback returns true
+    let subscriptionFound = true;
+    const valFound = res.response.some(element => {
+      if (userId === element.user_id || userAlias === element.user_alias) {
+        if (subscriptionId === element.subscription_id) {
+          subscriptionId = element.subscription_id;
+          subscriptionFound = true;
+          return true;
+        }
+        if (subscriptionAlias === element.subscription_alias) {
+          subscriptionAlias = element.subscription_alias;
+          subscriptionFound = true;
+          return true;
+        }
+        subscriptionFound = false;
+      }
+      return false;
+    });
+
+    // for a given userId, subscriptionId not found
+    if (!subscriptionFound) {
+      payload = constructPayload(message, createPayloadMapping);
+      payload = {
+        ...payload,
+        user_id: userId,
+        user_alias: userAlias
+      };
+      if (!currencyCodes.includes(payload.plan_currency.toLowerCase())) {
+        payload.plan_currency = null;
+      }
+      response.method = defaultPostRequestConfig.requestMethod;
+      response.endpoint = `${baseEndpoint}/v2/subscriptions`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      return response;
+    }
+
+    // userId and SubscriptionId is found
+    if (valFound) {
+      payload = constructPayload(message, updatePayloadMapping);
+      response.method = defaultPutRequestConfig.requestMethod;
+      response.endpoint = `${baseEndpoint}/v2/subscriptions/${subscriptionId ||
+        subscriptionAlias}`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      return response;
+    }
+  }
+
+  if (res.response.response.status !== 404) {
+    throw new CustomError(
+      "Failed to get subscription history for a user",
+      res.response.response.status
+    );
+  }
+  logger.debug("Failed to get subscription history for a user");
+
+  // userId and subscriptionId does not exist
+  // create new subscription for new user
+  payload = constructPayload(message, createPayloadMapping);
+  payload = {
+    ...payload,
+    user_alias: userAlias
+  };
+  if (!currencyCodes.includes(payload.plan_currency.toLowerCase())) {
```

Comment:
lets use a small oneline function instead of reusing the same code block in two places